src/main/java/com/ecore/roles/service/impl/MembershipsServiceImpl.java

```diff
@@ -0,0 +1,55 @@
+package com.ecore.roles.service.impl;
+
+import com.ecore.roles.exception.InvalidArgumentException;
+import com.ecore.roles.exception.ResourceExistsException;
+import com.ecore.roles.exception.ResourceNotFoundException;
+import com.ecore.roles.model.Membership;
+import com.ecore.roles.model.Role;
+import com.ecore.roles.repository.MembershipRepository;
+import com.ecore.roles.repository.RoleRepository;
+import com.ecore.roles.service.MembershipsService;
+import lombok.NonNull;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.UUID;
+
+import static java.util.Optional.ofNullable;
+
+@Log4j2
+@Service
+public class MembershipsServiceImpl implements MembershipsService {
+
+    private final MembershipRepository membershipRepository;
```

Comment:
It could be just called "repository", because it's inside of the membership service.